### ISSUE TITLE
SEPA Debit dispute date broken and messaging unclear

### ISSUE THREAD

**Issue by martynmjones** [Automattic/woocommerce-payments#3909]
### Describe the bug
If a customer opens a dispute for a SEPA Debit transaction within 8 weeks of placing the order then the dispute is immediately closed in their favor as expected but the dispute inside WooCommerce Payments reads `Respond by: Jan 1, 1970 - 1:00AM`

### To Reproduce

1.  Enable SEPA Debit as a payment method
2. Place order using a test account that will trigger a dispute i.e `GB28WEST12345698765434`
3. Go to `Payments > Disputes` and view dispute date details

### Screenshots

![Annotation on 2022-03-06 at 16-28-19](https://user-images.githubusercontent.com/40762232/156932236-c751e9e2-dd8b-4bb7-8bcb-348e3dab6ad0.png)

<img width="1282" alt="Markup 2022-03-06 at 16 37 01" src="https://user-images.githubusercontent.com/40762232/156932641-bf15b15f-1739-449f-9000-6b80f26e1b5e.png">

### Expected behavior
No date should be displayed. Additionally, the messaging states that evidence can be submitted whereas it's not possible for this dispute. Clearer messaging here that indicates why the dispute was immediately closed would avoid confusion for our users.

**Comment by theabhig** [comment]
Also reported in 5468258-zen

**Comment by neilmccreadie** [comment]
Also in 38089036-hc

**Comment by theabhig** [comment]
5825565-zen

**Comment by paulostp** [comment]
5829233-zen

**Comment by IliaDi** [comment]
5863244-zen

**Comment by sverleis** [comment]
5866103-zen

**Comment by jessy-p** [comment]
This issue impacts Deposits, Disputes, Capital Loans, so assigning to Helix (based on team responsibilities Pc2DNy-3z-p2) @haszari. Assigning as part of Gamma Triage process PcreKM-yM-p2.

Tagging as part of re-evaluating older issues in the backlog, please have a look and close if no longer relevant.

**Comment by haszari** [comment]
@pierorocca I see you've added this issye to Buyer Experiences Bug Project Board - https://github.com/orgs/Automattic/projects/880 - will a buyer experience team work on this? 

> If a customer opens a dispute for a SEPA Debit transaction within 8 weeks of placing the order then the dispute is immediately closed in their favor as expected but the dispute inside WooCommerce Payments reads Respond by: Jan 1, 1970 - 1:00AM

Looks like we're either inconsistent data from the Stripe API or we're somehow mishandling these disputes. 

#### Next steps
- Confirm if this is still happening:
  - Unclear from this issue as the tickets stop ~12 months ago, maybe was a temporary glitch. 
  - Testing / exploration to reproduce. I'm unfamiliar with testing SEPA payments (and disputes), hopefully there's a way to reproduce in test mode.

If it's still happening then we should fix.

**Comment by pierorocca** [comment]
Hey @haszari thanks for asking. The new project board is keying off of component labels to auto-identify issues most likely in our area. This issue has the `component: upe` label hence why it was added automatically. Would `component: disputes` be a more accurate label here?

**Comment by haszari** [comment]
> This issue has the component: upe label hence why it was added automatically. Would component: disputes be a more accurate label here?

Looks like the issue affects a dispute flow primarily. I'm not clear on the relevance of UPE for SEPA payments, presumably UPE allows SEPA.

**Comment by haszari** [comment]
> This issue impacts Deposits, Disputes, Capital Loans, so assigning to Helix (based on team responsibilities Pc2DNy-3z-p2) @haszari. Assigning as part of Gamma Triage process

Side note / question @jessy-p  – does the gamma triage process include adding component tag? I'm surprised this wasn't tagged with `disputes`.

The component tags are important for our workflow, this is how we pull issues into our board and prioritisation process (the comments are just a reminder / nudge). 

Added `disputes` label :)

**Comment by pierorocca** [comment]
Thanks @haszari. UPE is Stripe's UI element so it would be best applied to Checkout UI scenarios. I've advocated for a label cleanup and UPE is one that could be retired or collapsed into a more generic label like "Checkout" or "Checkout UI".

**Comment by haszari** [comment]
I see there are a few linked tickets (~5) so raising priority to `medium`. 

There are some questions here which would be good next steps:

- Find out how to reproduce this in test mode. Without that we will have a hard time fixing. 
- Find out how common this scenario is and get a better understanding of impact – could be `priority: high`.
- Find out more about this dispute flow and what it means. It's clearly different from a normal dispute flow, what's the normal flow for these SEPA disputes? What does Stripe recommend?

**Comment by dmallory42** [comment]
I came across an issue while on janitorial that I think may be related to this, but is presenting in a slightly different way.

In the issue (p1707422928616779-slack-C7U3Y3VMY) the merchant processed a SEPA payment which was later disputed/refunded by the buyer, but the transaction timeline on the merchant dashboard showed the `failed` payment event as a result of the dispute happening _before_ the `successful`  payment event (even though it actually happened afterwards). This confused the merchant making them think the payment was successful. 

My hunch here is that it's the same reason - because of how SEPA works, the dispute is closed automatically and thus no time should be recorded for it, however as a result we are defaulting towards showing it at the _beginning_ of the timeline, when it should actually display later.

Let me know if I should open up a new issue - I wanted to mention it here though because I do think it's likely to be related.

Screenshot with some info redacted:

<img width="1302" alt="Screenshot 2024-02-09 at 10 42 09" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/25f1c52d-cf77-4557-90d0-17876cd3b599">

**Comment by haszari** [comment]
Thanks for the screenshot & context @dmallory42 – very helpful, since we can't easily reproduce this in test mode.

**Comment by souravdebnath1986** [comment]
Putting priority as high. SEPA though has low usage has a high dispute activity rate of > 6%. Solving this could bring down the dispute rate.

**Comment by souravdebnath1986** [comment]
As a follow up from previous comment changed priority from high to low. Though the dispute activity rate is high, SEPA as a payment method is not supported. 
@pierorocca our [support documents ](https://woo.com/document/woopayments/payment-methods/additional-payment-methods/)show SEPA as an alternative payment method. Should this be updated and SEPA removed if it's no longer a supported payment method ? Also we still see < 100 active merchants per month offering SEPA. Should SEPA be disabled for these merchants as well ?

**Comment by pierorocca** [comment]
Thanks @souravdebnath1986. I've asked @csmcneill to remove the reference which links to a now removed page. I don't have the full history on the merchants that participated in the Beta and if they're being grandfathered or if the payment method is being sunset. Reaching out to a few folks.

**Comment by pierorocca** [comment]
@souravdebnath1986 it appears that only net new merchants cannot access SEPA and that some merchants are grandfathered in. p6q7sZ-bEI-p2#comment-34762 May be worth revisiting this to understand if the cost/benefit makes sense.

**Comment by csmcneill** [comment]
> I've asked @csmcneill to remove the reference which links to a now removed page.

Done!

**Comment by haszari** [comment]
> As a follow up from previous comment changed priority from high to low. Though the dispute activity rate is high, SEPA as a payment method is not supported.

Removing from sprint `next` as this is lower priority than we initially thought.

**Comment by fetenlakhal** [comment]
[8422167-ZD](https://a8c.zendesk.com/agent/tickets/8422167)

**Comment by francoishvz** [comment]
8550548-zd-a8c

**Comment by OmarFPG** [comment]
9696987-zd-a8c

**Comment by frosso** [comment]
The primary bug here - the epoch "Respond by: Jan 1, 1970" date and the misleading evidence-submission UI for auto-closed SEPA disputes - no longer occurs. 

The dispute UI was redesigned in #7047: all "Respond by" / evidence-submission rendering is now gated on dispute status. `renderDisputeDetails()` (summary/index.tsx:105) only renders the awaiting-response UI (which contains `DisputeDueByDate`) when `isAwaitingResponse(status)` is true, and that only matches `needs_response`/`warning_needs_response`. An auto-closed-in-buyer-favor dispute is `lost`/`charge_refunded`/`won`, so it routes to `DisputeResolutionFooter` with dedicated resolved copy and never reaches the epoch-date code.

The list column is similarly guarded in `smartDueDate` (disputes/index.tsx:158-172). SEPA is also no longer offered to new merchants, which shrinks the impact further.

The timeline event-ordering quirk @dmallory42 raised is a separate server-side concern - if it's still observed it should be filed on its own rather than holding this one open.

I think we can close this. Since disputes can't easily be reproduced in test mode, a quick confirmation would be welcome before we do.
